### ISSUE TITLE
Failed to install container on Ds 220+ Synology

### ISSUE THREAD

**Issue by albilia** [alfg/docker-nginx-rtmp#92]
Hello Alfred : )

i have been trying for a week to get nginx-rtmp-module and FFmpeg installed on Synology DS 220+ without success. 
i'm not familiar with the Linux system,
But can be run according to a guide with written commands. 
There is a situation  I did not understand the guide unfortunately, And I probably made nubes mistakes.
At first I tried to do the installation via ssh, but after the installation I didn't see any folders added so I could not associate the volume in the Docker container. 
After I failed with ssh I tried to install the container. But can not get to the page and in addition if I try to enter Bash through the terminal, the Bash screen just gets stuck. 
i  am convinced that for Linux people this is an action that takes 2-5 minutes :)
And this is the stage where I decided to give up and ask you for installation instructions. 
I enclose here the details of the JSON file created from your build installation for your reference.
PasteBin  link: https://pastebin.com/CZVSzu5F
i would be very happy if you could please look and tell me where I went wrong. Or give me the correct routes of the volumes. I do not know where I'm wrong. 

Thanks in advance
Nissim

**Comment by alfg** [comment]
Hi @albilia, 

Do you have the docker command or docker-compose.yml config you are using? You only need to mount a volume if you are overriding the nginx.conf with your own, or using ssl certs.

If mounting your own nginx.conf, make sure it's mounted to `/etc/nginx/nginx.conf.template` in the container.

**Comment by albilia** [comment]
Hi @alfg 

i'm using your container I found in a search in the Sinology Registry. 
![2021-02-26_03-10-45](https://user-images.githubusercontent.com/21350120/109244505-7f30fe00-77e7-11eb-9e26-ba4995d937b8.png)

Referring to the commands you wrote there and hoping it fits the Synology architecture, I logged into my DS220+ with SSH and put the commands you gave and refer to: "Pull docker image and run".
docker pull alfg / nginx-rtmp
docker run -it -p 1935: 1935 -p 8080: 80 --rm alfg / nginx-rtmp 
No errors were received and it would appear on the terminal screen that the command does indeed download and deploy files on the server. 
When I tried to search for the installation files I just performed, I didn't find them in the docker or any other folder. 
i hope this will answer your question regarding the docker-compose.yml file.
The second option was to manually move the "alfg / nginx-rtmp" you forked in github, the entire folder to the "Docker" folder in Synology. And from there ... build / guess how to build the volumes. 
From here all the questions started popping up. Did I install correctly, did I not install correctly, how do you browse the interface, does the 1935 port arrive at the end of the server address.
The information on how I build the volumes; such as which folder, and which "Mount Path"  refering to that folder - I am missing in the guide for nubes like me.
btw, Also in this [Third option] after installing your conainer "alfg / nginx-rtmp"  i downloaded from the docker registry, no actual installation is performed that can see it in the files.

**Comment by alfg** [comment]
Hey @albilia, 

I am not too familiar with the Synology platform, so I may be missing some context here. Once you run the `docker run..` command, are you able to load up the stats page at `http://localhost:8080/stats`? That should indicate if the application is running and ready to stream.

Also, running Docker doesn't really install any files anywhere since it's running a containerized application, so I'm not too sure which installation path you are looking for. Once pulling the image and running the command, the container should be ready on ports `1935` and `8080` for rtmp and http respectively. 

You can also try running this on your desktop OS (with Docker installed) to see if you are running the command correctly. 

Hope that helps.

**Comment by albilia** [comment]
Hi @alfg 

in order to start clean from the beginning, I executed the first command "docker pull alfg/nginx-rtmp" which passed smoothly without any problems. 
Steps of Second command:  
1. sudo docker container ls
2. sudo docker rm -f f63395741ec3 [I had to do this in order release port 1935]
3. sudo docker run -it -p 1935:1935 -p 8080:80 --rm alfg/nginx-rtmp
The result appears in the image.
Anyway, I do not want to bother you with this matter. I understand it has not yet been tried on Sinology so ... I will move on.
Many thanks Alfred :) 
![2021-02-26_10-35-12](https://user-images.githubusercontent.com/21350120/109276326-8c1d1400-781e-11eb-86f3-c626762a3c6f.png)

**Comment by alfg** [comment]
It seems like it's running in that screenshot, minus the SIGWINCH (screen size change) signals. Are you able to load `http://localhost:8080/stats` in the browser?

**Comment by albilia** [comment]
I tried but nothing happens.
The message displayed by the browser is: "This site cannot be accessed".
I enclose here a picture of the layout of the ports by containers. 
![1935](https://user-images.githubusercontent.com/21350120/109289460-ea51f300-782e-11eb-8357-9f17b6d4c63c.png)

**Comment by alfg** [comment]
Is this a separate device? Can you access by the device's IP?

**Comment by albilia** [comment]
Holy Molyyyyy....
this is the first time I have been able to log in with this address. I'm starting to think that the command: "sudo docker rm -f f63395741ec3" removed a container that contained same ports as 1935:8080
Is this the screen I'm supposed to get?
 
![Work](https://user-images.githubusercontent.com/21350120/109344135-fe214780-7876-11eb-9048-d79bb7298571.png)

**Comment by alfg** [comment]
Yep. So if you start streaming to `rtmp://192.168.2.20:1935/stream/hello`, then you should see more info getting populated. 

Then you can view the stream at: `http://192.168.2.20:8080/player.html?url=http://192.168.2.20:8080/live/hello.m3u8`

**Comment by albilia** [comment]
Now you spoke to me in Japanese ... LOL
I still do not know where to run the commands you just wrote. I really want to learn this whole thing. 
My goal is to take say streams from youtube and turn it into an m3u8 file to allow it to be streamed to iplaytv on Apple TV. 
Can it be done?

**Comment by alfg** [comment]
Ah, not too familiar with that kind of setup and pulling from Youtube. Can youtube push to a remote via rtmp?

**Comment by albilia** [comment]
i really hope I understood the purpose of Livestreamer; link : https://docs.livestreamer.io/
that Designed to function as a command-line utility that pipes video streams from various services into a video player.
the significance of this is huge for me. This means that I will be able to take any live video streaming from YouTube and broadcast it on the iptv app on Apple TV. Trying to understand how the matter works.
Oh and ... thank you so much for the help. Thanks to your guiding questions I was able to move forward and understand the steps. Thank you : ))
Since I am not familiar with the order of operations in the final treatment. It seems to me that the issue can be closed.

**Comment by alfg** [comment]
@albilia I see. Thanks for the context. However, the purpose of `nginx-rtmp` is to receive an `rtmp` stream as an ingest, and output HLS for HLS Players. `rtmp` is usually used with broadcasting software, such as https://obsproject.com/. 

I looked through the documentation of https://docs.livestreamer.io and https://streamlink.github.io, but didn't see any support for outputting `rtmp` as a piped stream. There _might_ be a way to do this with a combination of tools, such as https://github.com/ytdl-org/youtube-dl and ffmpeg. I haven't seen this type of setup with this project, though.

If you're doing this for personal use, you can actually just use OBS to capture a Youtube stream and send it to `nginx-rtmp` for HLS playback on the iptv app, but not sure if that's what you're looking for.

Basically, we just need an `rtmp` stream from the source. :)

**Comment by alfg** [comment]
Closing the issue for now since it's not an issue with this project. But feel free to respond with any questions and I'll continue to help or provide suggestions where I can. 🙂
